core/src/main/java/org/apache/iceberg/FilesTable.java

```diff
@@ -51,6 +52,20 @@ public static class FilesTableScan extends BaseFilesTableScan {
       super(table, schema, MetadataTableType.FILES, context);
     }
 
+    @Override
+    public TableScan useRef(String name) {
```

Comment:
I think in order to support other metadata tables which also have different schema than its base table schema, we might just pull this up from FilesTableScan to BaseMetadataTableScan instead so it will works with `.partitions` and `.entries` etc. Also I think BaseAllMetadataTableScan overrides this useRef method as well